lib/jellyfish_web/controllers/component_controller.ex

```diff
@@ -7,54 +7,37 @@ defmodule JellyfishWeb.ComponentController do
 
   action_fallback JellyfishWeb.FallbackController
 
-  def create(conn, %{"room_uuid" => room_uuid} = params) do
-    component_type =
-      params
-      |> Map.fetch!("component_type")
-      |> Component.validate_component_type()
-
-    component_options = Map.get(params, "component_options")
-
-    case {component_type, RoomService.find_room(room_uuid)} do
-      {:error, _} ->
-        conn
-        |> put_resp_content_type("application/json")
-        |> put_status(400)
-        |> json(%{errors: "Not proper component type"})
-
-      {{:ok, _component_type}, :not_found} ->
-        conn
-        |> put_resp_content_type("application/json")
-        |> put_status(400)
-        |> json(%{errors: "Room not found"})
-
-      {{:ok, component_type}, room_pid} ->
-        component = Room.add_component(room_pid, component_type, component_options)
-
-        conn
-        |> put_status(:created)
-        |> render("show.json", component: component)
+  def create(conn, %{"room_id" => room_id} = params) do
+    with {:ok, component_type_string} <- Map.fetch(params, "type"),
+         component_options <- Map.get(params, "options"),
+         {:ok, component_type} <- Component.validate_component_type(component_type_string) do
+      case RoomService.find_room(room_id) do
+        :not_found ->
```

Comment:
As I see it now, I think that `RoomService.find_room` should return `room_pid | {:error, :room_not_found}`.